src/library/scala/concurrent/Future.scala

```diff
@@ -413,8 +413,29 @@ trait Future[+T] extends Awaitable[T] {
    * @return        a `Future` with the result of the application of `f` to the results of `this` and `that`
    * @group Transformations
    */
-  def zipWith[U, R](that: Future[U])(f: (T, U) => R)(implicit executor: ExecutionContext): Future[R] =
-    flatMap(r1 => that.map(r2 => f(r1, r2)))(if (executor.isInstanceOf[BatchingExecutor]) executor else parasitic)
+  def zipWith[U, R](that: Future[U])(f: (T, U) => R)(implicit executor: ExecutionContext): Future[R] = {
```

Comment:
@lihaoyi Have you benchmarked this change?

Reply:
@viktorklang I haven't benchmarked the per-call overhead, but I have verified that the updated logic it stops slow futures from blocking other future failures from being immediately reported, which shaves 10s of seconds off some of my workflows.

Reply:
@lihaoyi It's important to make sure that it does not have an undue impact on other logic by reducing the baseline performance for everyone. you can probably use something like this: https://github.com/viktorklang/scala-futures/blob/master/benches/src/main/scala/scala/future/Benchmarks.scala#L399

Reply:
@viktorklang got it. Is there any convenient command I can run to run those? Or docs on how to run JMH suites? I'm not that familiar with the benchmarking infra in the scala/scala build

Reply:
`sbt "bench/jmh:runMain scala.concurrent.ZipWithFutureBenchmark"` is giving me `Error: Main method not found in class scala.concurrent.ZipWithFutureBenchmark`

Reply:
scratch that, `sbt "bench/jmh:run scala.concurrent.ZipWithFutureBenchmark"` seems to work